pandas/_typing.py

```diff
@@ -23,6 +23,7 @@
     Optional,
     Protocol,
     Type as type_t,
+    TypeAlias,
```

Comment:
It would be good to use TypeAlias: you can either skip it or import it dependent on the python version (introduced in py 3.10)

Reply:
yeah I think I figured out the fix. Let's see if the CI agrees with me or not.

Reply:
@twoertwein I'm not sure that `TypeAlias` should be introduced in this PR, because we have a lot of other places in `_typing.py` that could use `TypeAlias` .  If we introduce it here, then we're being inconsistent.

There should be a long term goal of using `TypeAlias` in `_typing.py`, but that should be in a separate PR.

I'll let you make the final decision on the matter.

Reply:
@tqa236 Can you please remove the TypeAlias for now. You are welcome to have a future PR that adds TypingAlias everywhere in _typing where appropriate.

Reply:
@twoertwein can you take a look again please? I reverted those commits